common/filestore_utils.py

```diff
@@ -27,16 +31,35 @@ def _using_gsutil():
     return experiment_filestore_path.startswith('gs://')
 
 
-if _using_gsutil():
-    from common import gsutil as filestore_utils_impl
-else:
-    # When gsutil is not used in the context, here it should use local_utils.
-    # TODO(zhichengcai): Implement local_filestore.py and import it here.
-    from common import gsutil as filestore_utils_impl
+global filestore_utils_impl
+filestore_utils_impl = None
+
+
+def get_filestore_utils_impl():
+    """Imports filestore_utils_impl dynamically."""
```

Comment:
This approach feels clean and even though I use a similar one in `database/utils.py` I think the more explicit `get_impl().cp` approach I showed you yesterday is better.

Reply:
I first tried that one (since that is easier), but failed yesterday. I guess that was because the assignment issues.
Let me do another check later.

Reply:
We can also use `return common.gsutil` in a function `get_impl()`.
Which one do you prefer? The current decorator one, or I rewrite to use `get_impl()`?
@lszekeres @jonathanmetzman

Reply:
Let's do `get_impl()` please.

Reply:
+1

Reply:
Done.